src/Icicle/Avalanche/Statement/Simp/Constructor.hs

```diff
@@ -80,7 +80,7 @@ constructor a_fresh statements
 
   goS env s
    = let env' = updateExpEnv s env
-         go   = goX env
+         go   = goX env'
```

Comment:
not 100% sure about this, need to make sure the updated environment from a non-recursive Let doesn't affect the let definition